### ISSUE TITLE
Private named instance fields are not removed from object spread

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#35943]
**TypeScript Version:**  3.8.0-dev.20191231

**Search Terms:**

**Code**

```ts
class C {
    #prop = 1;

    method(other: C) {
        const obj = { ...other };
        obj.#prop; // expected error, got no error
        const { ...rest } = other;
        rest.#prop; // expected error, got error
    }
}
```

**Expected behavior:**

Object spread should remove private named properties from the spreaded object type.

**Actual behavior:**

Private named properties are preserved on the spreaded type, accessing the property is not a type error. This causes a runtime error.

Note that object rest already works as expected.

**Playground Link:** https://www.typescriptlang.org/play/?ts=3.8.0-dev.20191231&ssl=10&ssc=2&pln=1&pc=1#code/MYGwhgzhAEDC0G8BQ1XQMQAcBOB7T0AvNAEwDcSKaAtgKYAuAFrgCYAUuTt2AXHAJSIqaNMFwA7CPWi4ARgCsiiaADo1nRt2gBfCiJFz5KrHkx79qMZOkJVa7LSk6lG7uYsOpxnPnc6k2kA

**Comment by RyanCavanaugh** [comment]
Thanks @ajafff !
